### PR TITLE
feat: extend charts with additional parameters

### PR DESCRIPTION
**Pull request by t3mi**
Extends both charts with following configurations:
- add optional securityContext object
- add affinity configuration
- add priorityClassName 
- add additional podLabels (e.g. `CostCenter=IT`)
- add configurable values for probes
  - failureThreshold
  - successThreshold
  - timeoutSeconds

### CODE REVIEW

File: docs/index.yaml [hunk 1]
```diff
@@ -3,7 +3,26 @@ entries:
   promitor-agent-resource-discovery:
```
Comment (tomkerkhove) [inline docs/index.yaml:3]: Can you revert this please? I'll do a release myself later on.

File: promitor-agent-resource-discovery/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v2
-version: 0.1.0
+version: 0.1.1
```
Comment (tomkerkhove) [inline promitor-agent-resource-discovery/Chart.yaml:2]: Can you revert this please? I'll do a release myself later on.

File: promitor-agent-scraper/Chart.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v2
-version: 2.1.0
+version: 2.1.1
```
Comment (tomkerkhove) [inline promitor-agent-scraper/Chart.yaml:2]: Can you revert this please? I'll do a release myself later on.

File: promitor-agent-scraper/values.yaml [hunk 4]
```diff
@@ -94,12 +94,24 @@ health:
     enabled: true
     verifyDependencies: false
     delay: 5
+    failureThreshold: 3
     interval: 5
+    successThreshold: 1
+    timeoutSeconds: 1
```
Comment (tomkerkhove) [inline promitor-agent-scraper/values.yaml:100]: How about this?
```suggestion
    interval: 5
    timeoutSeconds: 1
    thresholds:
      success: 1
      failure: 3
```

Comment (tomkerkhove) [reply promitor-agent-scraper/values.yaml:100]: If we change this, let's do it for all probes

Comment (tomkerkhove) [reply promitor-agent-scraper/values.yaml:100]: _I know this is the default for probes, but it's not exactly inline with the rest of the values_

**Review by tomkerkhove — COMMENTED** [review]
Thanks a ton for this contribution and sorry for the late review!

LGTM, just added a few comments

File: promitor-agent-resource-discovery/README.md [hunk 5]
```diff
@@ -81,12 +81,23 @@ their default values.
 | `health.readiness.enabled`  | Indication if readiness probes should be used | `true`            |   |
 | `health.readiness.verifyDependencies`  | Indication if readiness probes should verify if Promitor can interat with external dependencies. Do note that this will contact all dependencies which can have performance impact, cause throttling or cascading failures when consumed very often. | `false`            |   |
 | `health.readiness.delay`  | Amount of seconds to wait before probing the container to verify if it's ready | `5`            |   |
+| `health.readiness.failureThreshold`  | Number of retries after first failed check before container will be marked `Unready`  | `3` |
```
Comment (tomkerkhove) [inline promitor-agent-resource-discovery/README.md:84]:
```suggestion
| `health.readiness.thresholds.failure`  | Number of retries after first failed check before container will be marked `Unready`  | `3` |
```

File: promitor-agent-resource-discovery/README.md [hunk 6]
```diff
@@ -81,12 +81,23 @@ their default values.
...
 | `health.readiness.interval`  | Amount of seconds to wait before probing the container again to verify if it's ready after the last attempt | `5`            |   |
+| `health.readiness.successThreshold`  | Minimum consecutive successes for the probe to be considered successful after having failed | `1` |
```
Comment (tomkerkhove) [inline promitor-agent-resource-discovery/README.md:86]:
```suggestion
| `health.readiness.thresholds.success`  | Minimum consecutive successes for the probe to be considered successful after having failed | `1` |
```

File: docs/index.yaml (hunk 1 above)
Comment (t3mi) [reply docs/index.yaml]: Done. I've did that according to CONTRIBUTING.md doc :)

File: promitor-agent-scraper/README.md [hunk 7]
```diff
@@ -97,13 +97,24 @@ their default values.
 | `rbac.serviceAccount.annotations` | Service account annotations| `{}` |
 | `health.readiness.enabled`  | Indication if readiness probes should be used | `true`            |   |
 | `health.readiness.verifyDependencies`  | Indication if readiness probes should verify if Promitor can interat with external dependencies. Do note that this will contact all dependencies which can have performance impact, cause throttling or cascading failures when consumed very often. | `false`            |   |
-| `health.readiness.delay`  | Amount of seconds to wait before probing the container to verify if it's ready | `5`            |   |
-| `health.readiness.interval`  | Amount of seconds to wait before probing the container again to verify if it's ready after the last attempt | `5`            |   |
+| `health.readiness.delay`  | Amount of seconds to wait before probing the container to verify if it's ready | `5` |
+| `health.readiness.failureThreshold`  | Number of retries after first failed check before container will be marked `Unready`  | `3` |
```
Comment (tomkerkhove) [inline promitor-agent-scraper/README.md:101]:
```suggestion
| `health.readiness.thresholds.failure`  | Number of retries after first failed check before container will be marked `Unready`  | `3` |
```

File: promitor-agent-scraper/README.md [hunk 8]
```diff
@@ -97,13 +97,24 @@ their default values.
...
+| `health.readiness.interval`  | Amount of seconds to wait before probing the container again to verify if it's ready after the last attempt | `5`            |
+| `health.readiness.successThreshold`  | Minimum consecutive successes for the probe to be considered successful after having failed | `1` |
```
Comment (tomkerkhove) [inline promitor-agent-scraper/README.md:103]:
```suggestion
| `health.readiness.thresholds.success`  | Minimum consecutive successes for the probe to be considered successful after having failed | `1` |
```

File: promitor-agent-scraper/Chart.yaml (hunk 3 above)
Comment (t3mi) [reply promitor-agent-scraper/Chart.yaml]: Reverted.

**Review by tomkerkhove — APPROVED** [review]

File: promitor-agent-scraper/values.yaml (hunk 4 above)
Comment (t3mi) [reply promitor-agent-scraper/values.yaml]: Don't have any objections, changed.

File: docs/index.yaml (hunk 1 above)
Comment (tomkerkhove) [reply docs/index.yaml]: My bad, I should remove that

**Comment by tomkerkhove** [comment]
Thanks a ton @t3mi!

File: promitor-agent-resource-discovery/README.md (hunk 5 above)
Comment (t3mi) [reply promitor-agent-resource-discovery/README.md]: Sorry missed another commit. Now it should be intact.

**Pull request merged**